source/JsonObject.cpp

```diff
@@ -16,26 +16,24 @@ namespace Aws
     {
         JsonObject::JsonObject() : m_wasParseSuccessful(true) { m_value = nullptr; }
 
-        JsonObject::JsonObject(cJSON *value)
-            : m_value(cJSON_Duplicate(value, 1 /* recurse */)), m_wasParseSuccessful(true)
+        JsonObject::JsonObject(aws_json_value *value)
+            : m_value(aws_json_value_duplicate(value)), m_wasParseSuccessful(true)
         {
         }
 
         JsonObject::JsonObject(const String &value) : m_wasParseSuccessful(true)
         {
-            const char *return_parse_end;
-            m_value = cJSON_ParseWithLengthOpts(value.c_str(), value.length(), &return_parse_end, 0);
+            m_value = aws_json_value_new_from_string(ApiAllocator(), aws_byte_cursor_from_c_str(value.c_str()));
 
-            if (m_value == nullptr || cJSON_IsInvalid(m_value) == 1)
+            if (m_value == nullptr || aws_json_value_is_null(m_value) == true)
```

Comment:
I don't think the `aws_json_value_is_null()` check is necessary anymore?

I believe the contract of `aws_json_value_new_from_string()` is that it will return NULL or a valid aws_json_value that is a string

Reply:
I checked and it does return `null` on an error, so you are right that we do not need to check that anymore. It relies on `CJSON.Parse` catching the error and returning NULL, but any error path in `CJSON.Parse` does return `null` and not an null/empty CJSON object.